sgeop/artifacts.py

```diff
@@ -1420,3 +1404,11 @@ def is_dangle(edgelines):
     ).sum(axis=1)
 
     return (first_sum == 1) | (last_sum == 1)
+
+
+def line_segments(line: shapely.LineString) -> np.ndarray:
+    """Explode a linestring into constituent pairwise coordinates."""
+    xys = np.array(line.coords)
```

Comment:
```suggestion
    xys = shapely.get_coordianates(line)
```

This is about 3x faster.

Reply:
```suggestion
    xys = shapely.get_coordinates(line)
```

Reply:
> This is about 3x faster.

I was using that before and somehow got it swip-swapped out.